cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java

```diff
@@ -0,0 +1,281 @@
+ */
+
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.strimzi.operator.cluster.KafkaVersionTestUtils;
+import io.strimzi.operator.cluster.model.KafkaVersion;
+import io.strimzi.test.TestUtils;
+import org.apache.kafka.clients.admin.AlterConfigOp;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.common.config.ConfigResource;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+public class KafkaBrokerConfigurationDiffTest {
+
+    private static final KafkaVersion.Lookup VERSIONS = KafkaVersionTestUtils.getKafkaVersionLookup();
+    private static final String KAFKA_VERSION = "2.5.0";
+    KafkaVersion kafkaVersion = VERSIONS.version(KAFKA_VERSION);
+    private int brokerId = 0;
+
+    private String getDesiredConfiguration(List<ConfigEntry> additional) {
+        try (InputStream is = getClass().getClassLoader().getResourceAsStream("desired-kafka-broker.conf")) {
+            String desiredConfigString = TestUtils.readResource(is);
+
+            for (ConfigEntry ce : additional) {
+                desiredConfigString += "\n" + ce.name() + "=" + ce.value();
+            }
+
+            return desiredConfigString;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return "";
+    }
+
+    private Config getCurrentConfiguration(List<ConfigEntry> additional) {
+        List<ConfigEntry> entryList = new ArrayList<>();
+
+        try (InputStream is = getClass().getClassLoader().getResourceAsStream("current-kafka-broker.conf")) {
+
+            List<String> configList = Arrays.asList(TestUtils.readResource(is).split(System.getProperty("line.separator")));
+            configList.forEach(entry -> {
+                String[] split = entry.split("=");
+                String val = split.length == 1 ? "" : split[1];
+                ConfigEntry ce = new ConfigEntry(split[0].replace("\n", ""), val, true, true, false);
+                entryList.add(ce);
+            });
+            for (ConfigEntry ce : additional) {
+                entryList.add(ce);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        Config config = new Config(entryList);
+        return config;
+    }
+
+    private void assertConfig(KafkaBrokerConfigurationDiff kcd, ConfigEntry ce) {
+        Collection<AlterConfigOp> brokerDiffConf = kcd.getConfigDiff().get(new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(brokerId)));
```

Comment:
I've seen `new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(brokerId))` so many times in this PR. It would be better if it was factored into a `Util.brokerConfigResource(brokerId)`.